sdk/core/azure-core/azure/core/tracing/ext/opencensus_wrapper.py

```diff
@@ -0,0 +1,169 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+from azure.core.tracing import AbstractSpan
+from opencensus.trace import execution_context
+from opencensus.trace import tracer as tracer_module, Span
+from opencensus.trace.propagation import trace_context_http_header_format
+from opencensus.trace.samplers import ProbabilitySampler
+from opencensus.trace.tracers.noop_tracer import NoopTracer
+try:
+    from opencensus.ext.azure.trace_exporter import AzureExporter
+except:
+    AzureExporter = False
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, Union
+
+
+class OpencensusWrapper(AbstractSpan):
+    """Wraps a given Opencensus Span so that it implements azure.core.tracing.AbstractSpan"""
+
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Span, str) -> None
+        """
+        If a span is not passed in, creates a new tracer. If the instrumentation key for Azure Exporter is given, will
+        configure the azure exporter else will just create a new tracer.
+
+        :param span: The opencensus span to wrap
+        :param name: The name of the opencensus span to create if a new span is needed
+        """
+        super(OpencensusWrapper, self).__init__(span, name)
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None and (tracer is None or isinstance(tracer, NoopTracer)):
+            azure_exporter_instrumentation_key = self._get_environ(
+                "APPINSIGHTS_INSTRUMENTATIONKEY"
+            )
+            prob = float(self._get_environ("AZURE_TRACING_SAMPLER") or 0)
+            if azure_exporter_instrumentation_key is not None and AzureExporter:
+                tracer = tracer_module.Tracer(
+                    exporter=AzureExporter(
+                        instrumentation_key=azure_exporter_instrumentation_key
+                    ),
+                    sampler=ProbabilitySampler(prob),
+                )
+            else:
+                tracer = tracer or tracer_module.Tracer(sampler=ProbabilitySampler(prob))
```

Comment:
Not knowing how this should work, it seems odd to me that when `tracer` is a `NoopTracer`, we prefer whatever `opencensus.trace.Tracer` gives us. Is there a scenario in which we should continue using the `NoopTracer`?

Reply:
This is a tricky situation that might change in the future. The only time we overwrite the default tracer - Noop or not - is if the user sets the exporter through the environment variable. This means that the user did not create their own trace. This means that the user is ok with us making the trace as the exporter should only be set during trace creation. We will use Noop Tracer in all other scenarios. 

Also in our case, we only create the tracer once when we have not already created a span.

Reply:
nvmd -> talked to opentelemetry and they were like they will take care of this. Got rid of this section.